lib/ddtrace/profiling/transport/http.rb

```diff
@@ -0,0 +1,124 @@
+require 'ddtrace/ext/runtime'
+require 'ddtrace/ext/transport'
+
+require 'ddtrace/runtime/container'
+
+require 'ddtrace/profiling/transport/http/builder'
+require 'ddtrace/profiling/transport/http/api'
+
+require 'ddtrace/transport/http/adapters/net'
+require 'ddtrace/transport/http/adapters/test'
+require 'ddtrace/transport/http/adapters/unix_socket'
+
+module Datadog
+  module Profiling
+    module Transport
+      # TODO: Consolidate with Dataog::Transport::HTTP
+      # Namespace for HTTP transport components
+      module HTTP
+        module_function
+
+        # Builds a new Transport::HTTP::Client
+        def new(&block)
+          Builder.new(&block).to_transport
+        end
+
+        # Builds a new Transport::HTTP::Client with default settings
+        # Pass a block to override any settings.
+        def default(options = {})
+          new do |transport|
+            transport.headers default_headers
+
+            # Configure adapter & API
+            if options[:site] && options[:api_key]
+              configure_for_agentless(transport, options)
+            else
+              configure_for_agent(transport, options)
+            end
```

Comment:
Do we still want to support agentless @GaurabAryal?

Reply:
no need for now but we should design in a way to backport it down the road

Reply:
I will leave this for now (as the feature works), but we won't advertise/describe how to use this mode, if that's a workable interim solution @GaurabAryal .